core/sql/src/main/java/org/trafodion/sql/udr/TupleInfo.java

```diff
@@ -1079,19 +1230,48 @@ else if (t.getByteLength() == 8)
             row_.putLong(t.getDataOffset(), val);
             break;
 
+        case TINYINT_UNSIGNED:
+            if (val < 0 || val > Byte.MAX_VALUE)
+                if (val > (2 * Byte.MAX_VALUE + 1))
+                    throw new UDRException(
+                        38900,
+                        "Overflow/underflow when assigning to TINYINT UNSIGNED type");
+                else
+                    // use the signed value that has the same bit
+                    // pattern as the desired unsigned value, since
+                    // Java doesn't support "unsigned" basic types
+                    val = val + Byte.MIN_VALUE + Byte.MIN_VALUE;
```

Comment:
I suppose this is basic arithmetic. If the comment could please include an example that will help. Is Byte.MIN_VALUE = -127. If yes, how will adding it twice to val, give the same bit pattern as the unsigned type? I am quite sure there is no defect here, but that I am not understanding something obvious. Thanks.